angular/run/routes.run.js

```diff
@@ -1,17 +1,17 @@
-export function RoutesRun($rootScope, $state, $auth) {
+export function RoutesRun($state, $transitions) {
     'ngInject';
 
+    let requiresAuthCriteria = {
+        to: ($state) => $state.data && $state.data.auth
+    };
 
-    let deregisterationCallback =  $rootScope.$on("$stateChangeStart", function(event, toState) {
-
-        if (toState.data && toState.data.auth) {
-            /*Cancel going to the authenticated state and go back to the login page*/
-            if (!$auth.isAuthenticated()) {
-                event.preventDefault();
-                return $state.go('app.login');
-            }
+    let redirectToLogin = ['$auth', '$transition$', ($auth) => {
```

Comment:
is `$transitions` needed here?
And minification safe di works through ngInject.. so can you please move $auth to the top so that we avoid the duplication (between `'$auth'` and `$auth`)